gtda/mapper/utils/_visualization.py

```diff
@@ -176,102 +167,144 @@ def replace_summary_statistic(current_hovertext, new_statistic):
-    elif color_variable_kind == "callable":
-        color_data = color_variable(data)
-    elif color_variable_kind == "none":
-        if is_data_dataframe:
-            color_data = data.to_numpy()
+    if hasattr(color_features, "dtype") or hasattr(color_features, "dtypes"):
+        raise ValueError("`color_features` should not be a numpy array or "
+                         "pandas dataframe.")
+    elif hasattr(color_features, "fit_transform"):
+        color_features_kind = "fit_transformer"
+    elif hasattr(color_features, "transform"):
+        color_features_kind = "transformer"
+    elif callable(color_features):
+        color_features_kind = "callable"
+    elif color_features is None:
+        color_features_kind = "none"
+    else:  # Assume color_features is a selection of columns
+        color_features_kind = "else"
+
+    return color_features_kind
+
+
+def _validate_color_kwargs(graph, data, color_data, color_features,
+                           node_color_statistic, interactive=False):
+    if node_color_statistic is None:
+        node_color_statistic = np.mean
+
+    if color_data is None:
+        color_data = np.arange(len(data))
+
+    color_data_checked = check_array(color_data, ensure_2d=False, dtype=None)
+
+    # Simple duck typing to determine whether `color_data` is likely a pandas
+    # dataframe
+    is_color_data_dataframe = hasattr(color_data, "columns")
+
+    if len(color_data) != len(data):
+        raise ValueError("`color_data` and `data` must have the same length.")
+    if not is_color_data_dataframe:
+        color_data = color_data_checked.reshape((len(color_data), -1))
+
+    # Determine whether node_color_statistic is an array of node colors
+    is_node_color_statistic_ndarray = hasattr(node_color_statistic, "dtype")
+    is_node_color_statistic_callable = callable(node_color_statistic)
+
+    column_names_dropdown = None
+    if interactive:
+        if not is_node_color_statistic_callable:
+            raise ValueError("`node_color_statistic` must be a callable for "
+                             "interactive plots.")
+    else:
+        node_elements = graph.vs["node_elements"]
+        if is_node_color_statistic_ndarray:
+            node_color_statistic = check_array(node_color_statistic,
+                                               ensure_2d=False)
+            len_colors = len(node_color_statistic)
+            n_nodes = len(node_elements)
+            if len_colors != n_nodes:
+                raise ValueError(f"`node_color_statistic` must have as many "
+                                 f"entries as there are nodes in the Mapper "
+                                 f"graph. {len_colors} != {n_nodes} detected.")
+            if len(node_color_statistic.shape) > 1:
+                column_names_dropdown = range(node_color_statistic.shape[1])
+            else:
+                node_color_statistic = \
+                    node_color_statistic.reshape((len_colors, -1))
```

Comment:
Shouldn't it be something like
```
if len(node_color_statistic.shape) == 1:
     node_color_statistic = node_color_statistic.reshape((len_colors, -1))
column_names_dropdown = range(node_color_statistic.shape[1])
```
With the current implementation, `column_names_dropdown` is `None` if `node_color_statistic` was 1d. However, after this snippet, `node_color_statistic` is 2d, regardless of whether it was initially 1d or not.

Reply:
This is intentional.  There is no dropdown at all if there is only one set of colours, so `None` helps us remember that the notion of "column names for a dropdown menu" does not even apply in that case.

Reply:
Thanks for the clarification?

Reply:
Ok, I'll keep this thread open in case you want to think about it more/make another suggestion.